include/cassandra.h

```diff
@@ -2942,6 +2955,11 @@ cass_collection_append_decimal(CassCollection* collection,
                                size_t varint_size,
                                cass_int32_t scale);
 
+
+CASS_EXPORT CassError
+cass_collection_append_collection(CassCollection* collection,
```

Comment:
One thing to consider in this implementation: in v2, nested inner collections are encoded using v3 encoding. You may choose to make that explicit since you know we're putting a collection IN a collection. Some other drivers do not handle it gracefully.